apps/daimo-mobile/src/view/screen/OnboardingScreen.tsx

```diff
@@ -582,25 +602,179 @@ function UseExistingPage({
+  const [passkeyAccount, setPasskeyAccount] = useState<Account | undefined>(
+    undefined
+  );
+
+  const nextSlot = useMemo(
+    () =>
+      passkeyAccount
+        ? findUnusedSlot(
+            passkeyAccount.accountKeys.map((k) => k.slot),
+            SlotType.Mobile
+          )
+        : null,
+    [passkeyAccount]
+  );
+
+  const nonce = useMemo(
+    () => new DaimoNonce(new DaimoNonceMetadata(DaimoNonceType.AddKey)),
+    [pubKeyHex]
+  );
+
+  const sendFn = async (opSender: DaimoOpSender) => {
+    if (!pubKeyHex || !passkeyAccount || nextSlot === null)
+      throw new Error("[ACTION] passkey account not ready");
+    console.log(`[ACTION] adding device ${pubKeyHex}`);
+    return opSender.addSigningKey(nextSlot, pubKeyHex, {
+      nonce,
+      chainGasConstants: passkeyAccount.chainGasConstants,
+    });
+  };
+
+  const {
+    status: addDeviceStatus,
+    message: addDeviceMessage,
+    exec: addDeviceExec,
+  } = useSendAsync({
+    dollarsToSend: 0,
+    sendFn,
+    passkeyAccount,
+  });
+
+  const [restoreStatus, setRestoreStatus] = useState<{
+    status: ActStatus;
+    message: string;
+  }>({ status: "idle", message: "Restore from backup" });
+
+  const onRestoreBackup = async () => {
+    assert(pubKeyHex !== undefined);
+    console.log(`[ONBOARDING] restore backup attempt`);
+    setRestoreStatus({ status: "loading", message: "Requesting backup" });
+
+    const bChallenge = hexToBytes("0xdead");
+    const challengeB64 = base64.encode(bChallenge);
+    try {
+      const { accountName } = await requestPasskeySignature(
+        challengeB64,
+        env(daimoChain).passkeyDomain
+      );
+
+      const rpcFunc = env(daimoChain).rpcFunc;
+      const addr = await rpcFunc.resolveName.query({ name: accountName });
+
+      if (!addr) {
+        setRestoreStatus({
+          status: "error",
+          message: "Backup account not found",
+        });
+        return;
+      }
+
+      console.log(`[ONBOARDING] trying to restore ${accountName} ${addr}`);
+
+      const newAccount = createEmptyAccount(
+        {
+          name: accountName,
+          address: addr,
+          enclaveKeyName: defaultEnclaveKeyName,
+          enclavePubKey: pubKeyHex,
+        },
+        daimoChain
+      );
+
+      const syncedAccount = await hydrateAccount(newAccount);
+
+      setPasskeyAccount(syncedAccount);
+
+      // now add device will work if we use the passkey account
+      // We don't set the main account to the passkey account because
+      // we haven't yet added this device key to it.
+      setRestoreStatus({
+        status: "success",
+        message: `Successfully found account ${accountName}`,
+      });
+    } catch (e: any) {
+      console.error(e);
+      setRestoreStatus({ status: "error", message: e.message });
+      throw e;
+    }
+  };
+
   if (pubKeyHex === undefined) return null;
 
+  const deviceType = SlotType.Mobile; // TODO: detect and support desktop
```

Comment:
This would print `Add this Mobile to an existing account`
can use `env.deviceType` after #387

Reply:
done